### ISSUE TITLE
Building LNP from source

### ISSUE THREAD

**Issue by Ramblurr** [Lazy-Newb-Pack/LNP-Linux-legacy#4]
This github repo seems to be missing, well everything.

PyLNP isn't included, neither is the graphics pack repo. There isn't a build script for compiling the pack etc.

I would love to be able to hack on this project, to help with updates and such.

Could you add the dependencies as git submodules and write some documentation for how to fetch and compile all the other components to produce a working pack?

**Comment by BeauBouchard** [comment]
currently updating the package to 40.14, (even tho df just pushed 40.15 last night) :beer: 

 yes i will post more components of the package on the github repo.

**Comment by BeauBouchard** [comment]
Some of the files are missing because there are more than one version of the files for linux, for example DFhack previously was available in GCC 4.5.4 and 4.9.1 versions. It was my intention to branch the repo to post multiple versions. I was working on a script to download the correct versions of the components and then compiling them, however its become a futile process with DF updating so fast.  anyways, I will be posting more files for the 40.14 version, and branch it correctly now.

**Comment by Ramblurr** [comment]
Awesome! Would love to see that, especially the utilities and graphics.  A script for fetching dfhack would be fine I suspect.

You could always add the graphics repo as a git submodule?

Also, did you see [my post here](http://www.bay12forums.com/smf/index.php?topic=140966.msg5787772#msg5787772)? I'm not sure if it is a LNP, dfhack, or some other bug.

**Comment by fricy** [comment]
Graphics repo is [living here](https://github.com/fricy/DFgraphics) at the moment, looks like I don't have permission to fork it here. There's also an [LNP-core repo](https://github.com/PeridexisErrant/LNP-shared-core) but it needs a bit reorganizing so it can be added as a submodule into the LNP folder.

Instead of using platform specific scripts to fetch stuff I was looking at adding parts of [dulwich](https://github.com/jelmer/dulwich) as a module to PyLNP, but it's a bit over my head...

**Comment by Ramblurr** [comment]
Hey @fricy cools, thanks for the links.

I don't think the graphics repo needs to be forked, it can just be added to this repo as a submodule in the `graphics` directory.

What reorganizing needs to be done in LNP-core? That should be pretty simple.

A good first step in getting the platform specific stuff in the repo would be simple README files that document how/where to get the tools, and how to extract the archives into the relevant dir. Automation can come later.

**Comment by BeauBouchard** [comment]
@fricy sorry, didn't realize you guys can't make repos. I added you and @PeridexisErrant  to owners, I also forked your graphics repo, and the LNP-core repo, it would be great to use sub-modules to update the linux repo. 

@Ramblurr I am going to add some documentation today of "how to build" the package from compilation. Its a simple process that isn't documented well (or documented on each utilities page) that should be all in one spot.

**Comment by Ramblurr** [comment]
@BeauBouchard Cool, looking forward to that documentation.

Once you get it up, I can help with new releases. 

Also, what do you think about adding [the graphics repo](https://github.com/fricy/DFgraphics) and [LNP-core](https://github.com/PeridexisErrant/LNP-shared-core) as as a sub-module ?

**Comment by BeauBouchard** [comment]
I don't think we will be doing sub-modules for the time being, but  I documented some of the process including the compile [instructions in the wiki.](https://github.com/Lazy-Newb-Pack/Lazy-Newb-Pack-Linux/wiki/Package-update-Instructions)
